src/ledger/insert_shred.zig

```diff
@@ -635,21 +684,28 @@ pub const ShredInserter = struct {
     ) !ArrayList(CompletedDataSetInfo) {
         const slot = shred.fields.common.slot;
         const shred_index: u64 = @intCast(shred.fields.common.index);
+        // QUESTION: Why set this back to a union when we are in a function that deals with
```

Comment:
duplicate shreds can be either coding or data shreds, so we convert it to a union to add it to the arraylist (should probably moved to be defined right above where it is used)